library/Vanilla/Formatting/Html/DomUtils.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @author Dani M. <dani,m@vanillaforums.com>
+ * @author Isis Graziatto <isis.g@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting\Html;
+
+use DOMDocument;
+
+/**
+ * Class for stripping images and truncating text from a Dom.
+ */
+final class DomUtils {
+
+    /** @var array */
+    const EMBED_CLASSES = ['js-embed', 'embedResponsive', 'embedExternal', 'embedImage', 'VideoWrap', 'iframe'];
+
+    /**
+     * Remove embeds from the dom.
+     *
+     * @param DOMDocument $dom
+     * @param array $embedClasses
+     */
+    public static function stripEmbeds(DOMDocument $dom, array $embedClasses = self::EMBED_CLASSES): void {
+        $xpath = new \DomXPath($dom);
+        foreach ($embedClasses as $key => $value) {
+            $xpathQuery = $xpath->query(".//*[contains(@class, '$embedClasses[$key]')]");
```

Comment:
What about `iframe` and `video` tags?

Reply:
Dani and I took a look and we couldn't find any cases where `iframe` or `video` tags are actually embedded in the discussions. It's all `span` and `div` tags with specific CSS classes flagging that those elements are embeds.

Reply:
Right, but as a library function don't you think those should be stripped? Many of our old embeds to embed iframes I do believe.

Reply:
an iframe tag gets stripped when passed Gdn::formatService()->renderHTML()
As for video, i'll have to add that.

Reply:
I've added a check for video tag and a test for it.